sdk/go/pulumi/resource.go

```diff
@@ -467,6 +467,9 @@ type ResourceOptions struct {
 	// replacements.
 	ReplaceOnChanges []string
 
+	// if set, the engine will diff this with the last recorded value, and trigger a replace if they are not equal.
+	ReplacementTrigger Input
```

Comment:
I'd love to see a note on what the semantics are for `ReplacementTrigger` going from set to unset and vice versa. 

For example, what happens when you set `ReplacementTrigger` to a non-nil value for the first time; does it replace?

Reply:
I'd think it would make sense for the first set of the trigger value to not do a replace. But thats an engine not SDK concern.